proposal.tex

```diff
@@ -37,6 +48,17 @@ \subsection{BPS Batch }\label{sec:bpsbatch}
 This would also cover DMS-REQ-0119, DMS-REQ-0125, DMS-REQ-0128, DMS-REQ-0123, DMS-REQ-0127.
 
 
+\subsubsection{Non Rubin code} \label{sec:othercode}
+Users may wish to run arbitrary code on Rubin images.
+It should be possible to provide a butler recipe to get an image or set of images to the users space and allow them to run any code which takes FITS images.
+Done in a container this could be done for a large volume of images - it would have to get past  the resource allocation committee of course.
+This would require custom work by the user and would not be a general facility.
+Run in this mode little provenance would be kept.  (See \secref{sec:gen3})
+
+Some provenance could be garnered by ingesting the generated files which could at least give the execution graph of their creation.
```

Comment:
"The shim could be extended with basic support for ingesting output files via the Butler back into the user's personal space; this would provide some elementary provenance, including at least the execution graph of the files' creation."

Reply:
I think we have to require they use the datastore templates for file-naming conventions otherwise we won't know which file goes with which dataId.